src/PowerShellEditorServices/Services/PowerShell/Host/InternalHost.cs

```diff
@@ -0,0 +1,775 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.PowerShell.EditorServices.Hosting;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Console;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Context;
+using Microsoft.PowerShell.EditorServices.Services.PowerShell.Runspace;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation.Host;
+using SMA = System.Management.Automation;
+
+namespace Microsoft.PowerShell.EditorServices.Services.PowerShell.Host
+{
+    using Microsoft.PowerShell.EditorServices.Services.PowerShell.Debugging;
+    using Microsoft.PowerShell.EditorServices.Services.PowerShell.Execution;
+    using Microsoft.PowerShell.EditorServices.Services.PowerShell.Utility;
+    using Microsoft.PowerShell.EditorServices.Utility;
+    using System.IO;
+    using System.Management.Automation;
+    using System.Management.Automation.Runspaces;
+    using System.Reflection;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    internal class InternalHost : PSHost, IHostSupportsInteractiveSession, IRunspaceContext
+    {
+        private const string DefaultPrompt = "PSIC> ";
+
+        private static readonly string s_commandsModulePath = Path.GetFullPath(
+            Path.Combine(
+                Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location),
+                "../../Commands/PowerShellEditorServices.Commands.psd1"));
+
+        private readonly ILoggerFactory _loggerFactory;
+
+        private readonly ILogger _logger;
+
+        private readonly ILanguageServerFacade _languageServer;
+
+        private readonly HostStartupInfo _hostInfo;
+
+        private readonly BlockingConcurrentDeque<ISynchronousTask> _taskQueue;
```

Comment:
@PaulHigin

Reply:
Thanks.  I think I need to clone this repo so I can better see how the changes are used.  Searching within this PR instance does not provide enough context.